src/parse.jl

```diff
@@ -212,7 +212,7 @@ end
 function parse_di(di_name, rig, sample_rate::HasInverseTimeUnits)
     insigs = getdigital(getinputs(rigtemplate(rig; sample_rate = sample_rate)))
     nsamples = filesize(di_name) #each sample is one byte
-    f = open(di_name, "r")
+    f = open(di_name, "r+")
```

Comment:
What's this for? Someone could overwrite the file just by assigning values to `A`. Is that desired? `parse_di` seems like a pretty innocuous a name for something that might damage the file.

Reply:
Yes you are right. But, I met "ERROR: ArgumentError: unable to increase file size to 697200 due to read-only permissions" error in Windows system. So, I tried changing the option.

Stacktrace:
 [1] mmap(::IOStream, ::Type{Array{UInt64,1}}, ::Tuple{Int64}, ::Int64; grow::Bool, shared::Bool) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Mmap\src\Mmap.jl:213
 [2] mmap(::IOStream, ::Type{BitArray}, ::Tuple{Int64,Int64}, ::Int64; grow::Bool, shared::Bool) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Mmap\src\Mmap.jl:292
 [3] mmap at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Mmap\src\Mmap.jl:290 [inlined] (repeats 2 times)
 [4] parse_di(::String, ::SubString{String}, ::Unitful.Quantity{Int64,�^-1,Unitful.FreeUnits{(s^-1,),�^-1,nothing}}) at C:\Users\kdw76\.julia\dev\ImagineInterface\src\parse.jl:216
 [5] parse_di(::String, ::Array{String,1}, ::Array{SubString{String},1}, ::SubString{String}, ::Unitful.Quantity{Int64, �^-1,Unitful.FreeUnits{(s^-1,),�^-1,nothing}}; load_unused::Bool) at C:\Users\kdw76\.julia\dev\ImagineInterface\src\parse.jl:190
 [6] parse_di(::String, ::Dict{String,Any}; load_unused::Bool) at C:\Users\kdw76\.julia\dev\ImagineInterface\src\parse.jl:184
 [7] parse_di at C:\Users\kdw76\.julia\dev\ImagineInterface\src\parse.jl:169 [inlined]
 [8] parse_di(::String; imagine_header::String, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at C:\Users\kdw76\.julia\dev\ImagineInterface\src\parse.jl:158

Reply:
Interesting? How close is 697200 to the actual size? Maybe this is just a round-up vs round-down thing on the [page size](https://en.wikipedia.org/wiki/Page_%28computer_memory%29)? If so, in my view it would be better to pretend the file is a bit smaller and leave the permissions read-only.

It's not that there's never a reason to open with `r+` permissions (for example, if you're deliberately writing a file), but I worry that this could be a pretty dangerous place to do it from.

Reply:
Actual size is 697,199.
So, do you suggest that we assign a little smaller memory map?
For example, 

```julia
 nsamples = filesize(di_name) #each sample is one byte
 f = open(di_name, "r")
rd_nsamples = floor(Int, nsamples/10)*10 # make last digit zero
 A = Mmap.mmap(f, BitArray, (8,rd_nsamples))
```

Reply:
Yeah, but use 64 rather than 10: https://github.com/JuliaLang/julia/blob/bd42c0a04612ed675f8471f2b9c7d3735b05c0ee/base/bitarray.jl#L25.